realm/realm-library/src/main/cpp/io_realm_internal_Collection.cpp

```diff
@@ -433,3 +433,19 @@ Java_io_realm_internal_Collection_nativeGetMode(JNIEnv *env, jclass, jlong nativ
     return -1; // Invalid mode value
 }
 
+JNIEXPORT jlong JNICALL
+Java_io_realm_internal_Collection_nativeCreateResultsFromBacklinks(JNIEnv *env, jclass, jlong shared_realm_ptr, jlong row_ptr, jlong src_table_ptr, jlong src_col_index) {
+    TR_ENTER_PTR(row_ptr)
+    Row* row = ROW(row_ptr);
+    if (!ROW_VALID(env, row))
+        return -1;
+    try {
+        Table* src_table = TBL(src_table_ptr);
+        TableView backlink_view = row->get_table()->get_backlink_view(row->get_index(), src_table, src_col_index);
+        auto shared_realm = *(reinterpret_cast<SharedRealm*>(shared_realm_ptr));
+        Results results(shared_realm, std::move(backlink_view));
+        auto wrapper = new ResultsWrapper(results);
```

Comment:
Thanks for catching this!  I really don't know enough about C++ memory management to know what is safe. I though, from looking at: 
```Java_io_realm_internal_Collection_nativeCreateResults```
that the purpose of the wrapper was to manage that exact problem. Please advise.